app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/services/ApplicationService.scala

```diff
@@ -61,10 +61,10 @@ class ApplicationService @Inject()(
       case Some(request) => {
         thirdPartyApplicationConnector.addTermsOfUseAcceptance(application.id, request).map(_ match {
           case Left(upstreamErrorResponse: UpstreamErrorResponse) => Left(upstreamErrorResponse.message)
-          case Right(_) => Right()
+          case Right(_) => Right(())
```

Comment:
This is interesting, what is the significance of the double brackets?

Reply:
That is the "Unit" return type wrapped in a Right.  Without it, Scala will infer a unit response but will warn that this is not the best.  Had it not been a Unit type it would have been more obvious what was wrong.